File: examenes/ExamenDialogos/app/src/main/java/com/system/develop/examendialogos/DialogoHora.java
package com.system.develop.examendialogos;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;

import java.util.Calendar;

public class DialogoHora extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        TimePickerDialog timePickerDialog = new TimePickerDialog(getActivity(), (TimePickerDialog.OnTimeSetListener) getActivity(),
                Calendar.getInstance().get(Calendar.HOUR_OF_DAY),
                Calendar.getInstance().get(Calendar.MINUTE),true);

        return timePickerDialog;
    }
}

File: Kotlin/T5/codigos2223/T5_Dalogos_RET/app/src/main/java/com/develop/t5_dalogos_ret/dialogs/DialogoHora.kt
package com.develop.t5_dalogos_ret.dialogs

import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.develop.t5_dalogos_ret.R
import java.util.Calendar

class DialogoHora: DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val calendario: Calendar = Calendar.getInstance();
        val hora = calendario.get(Calendar.HOUR)
        val minutos = calendario.get(Calendar.MINUTE)

        val dialogoHora = TimePickerDialog(requireContext(),requireContext() as OnTimeSetListener,hora,minutos,true)
        TimePickerDialog(requireContext(),
            R.style.CuadroColor,requireContext() as OnTimeSetListener,hora,minutos,true)

        return dialogoHora
    }
}